webapp/src/main/java/com/box/l10n/mojito/rest/repository/BranchStatisticWS.java

```diff
@@ -52,41 +44,70 @@ public class BranchStatisticWS {
+                    ifParamNotNull(deletedEquals(deleted))).and(
+                    ifParamNotNull(empty(empty))).and(
+                    ifParamNotNull(createdBefore(createdBefore))).and(
+                    ifParamNotNull(createdAfter(createdAfter)))
+            , pageable);
 
-    // 2. Hydrate BranchStatistic entities for JSON response using eager loading to avoid N+1
-    // queries
-    List<BranchStatistic> branchStatistics =
-        branchStatisticRepository.findByIdIn(branchStatisticIds.getContent(), pageable.getSort());
+    // Retrieve the Branch Statistic Ids which have TmTextUnit count less than the 'totalCountLte'
+    Page<Long> branchStatisticIdsWithLessThan = sparseBranchStatisticRepository.findAllWithIdOnly(where(
+                    ifParamNotNull(createdByUserNameEquals(createdByUserName))).and(
+                    ifParamNotNull(branchEquals(branchId))).and(
+                    ifParamNotNull(branchNameEquals(branchName))).and(
+                    ifParamNotNull(search(search))).and(
+                    ifParamNotNull(deletedEquals(deleted))).and(
+                    ifParamNotNull(empty(empty))).and(
+                    ifParamNotNull(createdBefore(createdBefore))).and(
+                    ifParamNotNull(createdAfter(createdAfter))).and(
+                    totalCountLessThanOrEqualsTo(totalCountLte))
+            , pageable);
 
+    // Retrieve the branch statistics where the total text unit count is greater than the limit,
+    // by filtering the branchStatisticIds less than the limit.
+    Page<BranchStatistic> branchStatisticsWithoutTextUnits = branchStatisticRepository.findAll(where(
+                    ifParamNotNull(createdByUserNameEquals(createdByUserName))).and(
+                    ifParamNotNull(branchEquals(branchId))).and(
+                    ifParamNotNull(branchNameEquals(branchName))).and(
+                    ifParamNotNull(search(search))).and(
+                    ifParamNotNull(deletedEquals(deleted))).and(
+                    ifParamNotNull(empty(empty))).and(
+                    totalCountGreaterThan(totalCountLte))
+            , pageable);
+
+    // Set the BranchTextUnitStatistics to null as the branch has a large number of text Units.
+    branchStatisticsWithoutTextUnits.forEach(
+            branchStatistic -> {branchStatistic.setBranchTextUnitStatistics( null );});
+
+
+    // 2. Hydrate BranchStatistic entities for JSON response using eager loading to avoid N+1 queries
+    List<BranchStatistic> branchStatistics = branchStatisticRepository.findByIdIn(
+            branchStatisticIdsWithLessThan.getContent(), pageable.getSort());
+
+    // Merge the two lists:
+    branchStatistics.addAll(branchStatisticsWithoutTextUnits.getContent());
     PageImpl<BranchStatistic> page =
-        new PageImpl<>(
-            branchStatistics,
-            branchStatisticIds.getPageable(),
-            branchStatisticIds.getTotalElements());
+            new PageImpl<>(
```

Comment:
Looks like the formatting tool wasn't run locally and IntelliJ wasn't configured with the Google Java styleguide - could you run a "mvn compile" locally to fix?